src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2774,13 +2778,18 @@ def query(self, filters=None, options=None):
             if k8s_config['dataset']:
                 filters.append(['id', '!^', f'{k8s_config["dataset"]}/'])
 
+        extra = copy.deepcopy(options.get('extra', {}))
+        retrieve_children = extra.get('retrieve_children', True)
         return filter_list(
             self.__transform(self.middleware.call_sync(
-                'zfs.dataset.query', zfsfilters, {'extra': {'flat': options.get('extra', {}).get('flat', True)}})
+                'zfs.dataset.query', zfsfilters, {
+                    'extra': {'flat': extra.get('flat', True), 'retrieve_children': retrieve_children}
```

Comment:
To be consistent, why not create another variable called `flat = extra.get('flat', True)` like you did with retrieve_children?

Reply:
I can do that but that can grow :) Motivation was to not have an extra line due to line being too long ( 120 > ). If you think i should, i can :p

Reply:
It's me just being OCD mostly, but I like the consistency of pulling out the extra arguments in their own variables. Looks cleaner, but obviously not needed. I'll leave it up to you. 😄